mne/preprocessing/maxwell.py

```diff
@@ -505,3 +553,113 @@ def _update_sss_info(raw, origin, int_order, ext_order, nsens):
+    """Helper to replace sensor geometry information"""
+
+    # Replace sensor locations (and track differences) for fine calibration
+    ang_shift = np.zeros((info['nchan'], 3))
+    for cal_chan in cal_chans:
+        ch_ind = info['ch_names'].index(cal_chan['ch_name'])
+
+        # Loop over vectors in X, Y, and Z directions
+        for ang_i, (i1, i2) in enumerate(zip(range(3, 10, 3),
+                                             range(6, 13, 3))):
+            v1 = cal_chan['loc'][i1:i2]
+            v2 = info['chs'][ch_ind]['loc'][i1:i2].astype(float)
+
+            cos_ang = np.dot(v1, v2) / (np.linalg.norm(v1) *
+                                        np.linalg.norm(v2))
+            ang_shift[ch_ind, ang_i] = cos_ang
+
+        # Adjust channel orientation with those from fine calibration
+        info['chs'][ch_ind]['loc'] = cal_chan['loc']
+
+    ang_shift[ang_shift > 1.] = 1.  # Deal with unreasonable values
+    ang_shift = np.arccos(ang_shift) * (180. / np.pi)  # Convert to degrees
+
+    # Log quantification of sensor changes
+    logger.info('Fine calibration adjusted coil positions by (mean ± std in '
+                'degrees): %0.5f ± %0.5f (max: %0.5f)' %
+                (np.mean(ang_shift), np.std(ang_shift),
+                 np.max(np.abs(ang_shift))))
+
+    return info
+
+
+def _sss_basis_point(origin, info, int_order, ext_order, imbalance):
+    """Compute multipolar moments for point-like magnetometers (in fine cal)"""
+
+    # Construct 'coils' with r, weights, normal vecs, # integration pts, and
+    # channel type.
+    if imbalance.ndim == 1:
+        imbalance = imbalance[:, np.newaxis]
+    if imbalance.shape[1] not in [1, 3]:
+        raise ValueError('Must have 1 (x) or 3 (x, y, z) point-like ' +
+                         'magnetometers. Currently have %i' %
+                         imbalance.shape[1])
+
+    # Initialize internal multipolar moment space for point-like magnetometers
+    S_in_all = np.zeros((len(info['chs']), (int_order + 1) ** 2 - 1))
+    S_out_all = np.zeros((len(info['chs']), (ext_order + 1) ** 2 - 1))
+
+    # Coil_type values for x, y, z point magnetometers
+    pt_types = [FIFF.FIFFV_COIL_POINT_MAGNETOMETER_X,
+                FIFF.FIFFV_COIL_POINT_MAGNETOMETER_Y,
+                FIFF.FIFFV_COIL_POINT_MAGNETOMETER]
+
+    # Loop over all coordinate directions desired and create point mags
+    for dir_ind in np.arange(imbalance.shape[1]):
+
+        temp_info = deepcopy(info)
+        for ch in temp_info['chs']:
+            ch['coil_type'] = pt_types[dir_ind]
+        coils_add, _, _, _ = _prep_meg_channels(temp_info, accurate=True,
+                                                elekta_defs=True)
+
+        S_in_add, S_out_add = _sss_basis(origin, coils_add, int_order,
+                                         ext_order)
+        # Scale spaces by gradiometer imbalance
+        S_in_all += S_in_add * imbalance[:, dir_ind].reshape(-1, 1)
```

Comment:
the reshape is more vague / less common than doing `..., dir_ind][:, np.newaxis]`